src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java

```diff
@@ -43,26 +48,57 @@ public NewHandler(IndentationCheck indentCheck,
 
     @Override
     public void checkIndentation() {
-        final DetailAST type = getMainAst().getFirstChild();
-        if (type != null) {
-            checkExpressionSubtree(type, getIndent(), false, false);
+        final DetailAST firstChild = getMainAst().getFirstChild();
+        final DetailAST mainAst = getMainAst();
+
+        // if new is on the line start and it is not the part of assignment.
+        if (isOnStartOfLine(mainAst)
+                && !getMainAst().getParent().getParent().getText().equals(ASSIGNMENT)) {
+            final int columnNo = expandedTabsColumnNo(mainAst);
+            final IndentLevel level = getIndentImpl();
+
+            if (columnNo < level.getFirstIndentLevel()) {
+                logError(mainAst, "", columnNo, level);
+            }
+        }
+
+        if (firstChild != null) {
+            checkExpressionSubtree(firstChild, getIndent(), false, false);
         }
 
-        final DetailAST lparen = getMainAst().findFirstToken(TokenTypes.LPAREN);
+        final DetailAST lparen = mainAst.findFirstToken(TokenTypes.LPAREN);
         checkLeftParen(lparen);
     }
 
+    @Override
+    public IndentLevel getSuggestedChildIndent(AbstractExpressionHandler child) {
+        final IndentLevel childIndent;
+        if (child.getMainAst().getType() == TokenTypes.OBJBLOCK) {
+            childIndent = new IndentLevel(getIndent(), getBasicOffset());
+        }
+        else {
+            childIndent = new IndentLevel(getIndent(),
+                    getIndentCheck().getLineWrappingIndentation());
+        }
+        return childIndent;
```

Comment:
As for me, it will look a bit better with something like
```
final int offset;
if (child.getMainAst().getType() == TokenTypes.OBJBLOCK) {
            offset= getBasicOffset();
        }
        else {
            offset= getIndentCheck().getLineWrappingIndentation());
        }
return new IndentLevel(getIndent(), offset);
```